mixnet-utils/src/lib.rs

```diff
@@ -0,0 +1,69 @@
+use std::time::Duration;
+
+use rand::{rngs::StdRng, thread_rng, Rng, RngCore, SeedableRng};
+use rand_distr::{Distribution, Exp};
+
+/// Get a random interval from an exponential distribution.
+pub fn poisson_interval_sec(rate_per_min: f64) -> Duration {
+    // create an exponential distribution
+    let exp = Exp::new(rate_per_min).unwrap();
+    // generate a random value from the distribution
+    let interval_min = exp.sample(&mut thread_rng());
+    // convert minutes to seconds
+    Duration::from_secs_f64(interval_min * 60.0)
+}
+
+/// Get the mean interval in seconds for a Poisson.
+pub fn poisson_mean_interval_sec(rate_per_min: f64) -> Duration {
+    // the mean interval in seconds for a Poisson process
+    Duration::from_secs_f64(1.0 / rate_per_min * 60.0)
+}
+
+/// Fisher-Yates shuffling algorithm.
+pub struct FisherYates<R: SeedableRng>(std::marker::PhantomData<R>);
```

Comment:
We may be able to reuse the existing fisheryates in Carnot https://github.com/logos-co/nomos-node/blob/3f2905817a92f0e8714b273183b2c0515004ea7f/consensus/carnot-engine/src/overlay/membership.rs#L16